controller/backup_controller.go

```diff
@@ -413,153 +415,122 @@ func (bc *BackupController) validateBackingImageChecksum(volName, biName string)
 	return bi.Status.Checksum, nil
 }
 
-func (bc *BackupController) backupCreation(backupTarget *longhorn.BackupTarget, volume *longhorn.Volume, backup *longhorn.Backup) (skip bool, err error) {
-	log := bc.logger.WithFields(
-		logrus.Fields{
-			"backup":   backup.Name,
-			"vol":      volume.Name,
-			"snapshot": backup.Spec.SnapshotName,
-			"label":    backup.Spec.Labels,
-		},
-	)
-
-	currentState := backup.Status.State
+// backupCreation calls engine's snapshot backup API or sync current engine's snapshot backup progress/state
+// if the `spec.SnapshotName` is set and `status.lastSyncedAt` is unset.
+// note that:
+// - if the CR without the spec.SnapshotName, the sourcce of truth is the remote backup target
+// - if the CR with the spec.SnapshotName, there are two source of truth:
+//   - the 1st is the engine, longhorn manager calls engine's snapshot backup API and monitoring the longhorn engine reports progress/state
+//   - the 2nd is once the engine reports snapshot backup state Completed, the source of truth becomes the remote backup target
+func (bc *BackupController) backupCreation(backup *longhorn.Backup, volumeName string, backupTarget *longhorn.BackupTarget) error {
+	var volume *longhorn.Volume
+	var monitor *engineapi.BackupMonitor
+	existingBackupState := backup.Status.State
 	defer func() {
-		if currentState == backup.Status.State {
+		if volume == nil {
 			return
 		}
-		if err != nil {
+
+		if monitor != nil {
+			backupStatus := monitor.GetBackupStatus()
+
+			backup.Status.Progress = backupStatus.Progress
+			backup.Status.URL = backupStatus.URL
+			backup.Status.Error = backupStatus.Error
+			backup.Status.SnapshotName = backupStatus.SnapshotName
+			backup.Status.ReplicaAddress = backupStatus.ReplicaAddress
+			backup.Status.State = backupStatus.State
+		}
+		if existingBackupState == backup.Status.State {
+			return
+		}
```

Comment:
Do not return before printing out the error.

Reply:
We only want to emit an event and log when the backup status state change.
For example:
- New -> InProgress
- InProgress -> Completed
- New -> Error
- InProgress -> Error